### PR TITLE
DayThree.

### PR DESCRIPTION
**Pull request by thanikasomasundaram-t**
Assignment 3 review - > multiThreading

### CODE REVIEW

File: daythree/src/canthreadruntwotimes/ThreadRepeat.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+	
+	public static void main(String[] args) {
+		Thread one = new Thread(() -> {
+			System.out.println("thread called");
+		});
+		one.start();
+		
+		try {
+		one.start();
```
Comment (Adheesh-Sirius) [inline daythree/src/canthreadruntwotimes/ThreadRepeat.java:14]: Can you add your observations as comments at the bottom?

File: daythree/src/runningcharacters/RunningCharacters.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+public class RunningCharacters {
+
+	public static void main(String[] args) {
+		RunningCharacters rc = new RunningCharacters();
+		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
+
+		while(true) {
+			try {
+				int c = bf.read();
```
Comment (Adheesh-Sirius) [inline daythree/src/runningcharacters/RunningCharacters.java]: please prefer using elaborated variable names. We'd lose track of what `c` or `bf` mean if this function becomes complex

File: daythree/src/printMessage/PrintMessage.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class PrintMessage {
+	
+	public static void display(String message) {
+		for(int i = 0; i< 3; i++)
+			System.out.println("[ ------" + message + "------ ]");
+	}
+
```
Comment (Adheesh-Sirius) [inline daythree/src/printMessage/PrintMessage.java:11]: Here, the goal is to print the string `[------Message------]` with different threads. For instance, thread 1 should print '[', thread 2 should print '------', etc. Can you update the program accordingly?

File: daythree/src/oddEven/OddEven.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class OddEven {
+	
+	
+	public static void main(String[] args) {
+		ExecutorService es = Executors.newFixedThreadPool(2);
+
```
Comment (Adheesh-Sirius) [inline daythree/src/oddEven/OddEven.java:11]: Adding comments that explains your approach would be helpful

**Review by Adheesh-Sirius — CHANGES_REQUESTED** [review]

File: daythree/src/canthreadruntwotimes/ThreadRepeat.java (hunk 1 above)
Comment (thanikasomasundaram-t) [reply daythree/src/canthreadruntwotimes/ThreadRepeat.java:14]: Added the observation.

File: daythree/src/runningcharacters/RunningCharacters.java (hunk 2 above)
Comment (thanikasomasundaram-t) [reply daythree/src/runningcharacters/RunningCharacters.java]: Changed to more semantic name.

File: daythree/src/printMessage/PrintMessage.java (hunk 3 above)
Comment (thanikasomasundaram-t) [reply daythree/src/printMessage/PrintMessage.java:11]: tried to move each design message to each thread.

File: daythree/src/oddEven/OddEven.java (hunk 4 above)
Comment (thanikasomasundaram-t) [reply daythree/src/oddEven/OddEven.java:11]: Added comments to understand the flow.

**Review by Adheesh-Sirius — APPROVED** [review]

**Review by Adheesh-Sirius — APPROVED** [review]

**Pull request merged**